src/Tribe/REST/V1/Post_Repository.php

```diff
@@ -232,21 +232,20 @@ public function get_venue_data( $event_or_venue_id, $context = '' ) {
 		// Add the Global ID fields
 		$data = $this->add_global_id_fields( $data, $venue->ID );
 
-		/**
-		 * Filters the data that will be returned for a single venue.
-		 *
-		 * @param array   $data  The data that will be returned in the response.
-		 * @param WP_Post $event The requested venue.
-		 */
-		$data = apply_filters( 'tribe_rest_venue_data', $data, $venue );
+		$event = null;
+
+		if ( tribe_is_event( $event_or_venue_id ) ) {
+			$event = get_post( $event_or_venue_id );
+		}
 
 		/**
-		 * Filters the data that will be returned for an event venue.
+		 * Filters the data that will be returned for a single venue.
 		 *
-		 * @param array   $data  The data that will be returned in the response.
-		 * @param WP_Post $event The requested event.
+		 * @param array        $data  The data that will be returned in the response.
+		 * @param WP_Post      $venue The requested venue.
+		 * @param WP_Post|null $event The requested event, if event ID was used.
 		 */
-		$data = apply_filters( 'tribe_rest_venue_data', $data, get_post( $event_or_venue_id ) );
```

Comment:
FYI, this used to be a separate filter called `tribe_rest_event_venue_data`, so there were really two hooks. But then at some point the filter was mistakenly renamed as a result of a namespacing commit. Now it's problematic for hooking into since the hook runs twice and could have varied second argument data.